test/functional/zmq_test.py

```diff
@@ -13,14 +13,39 @@
                                  hash256,
                                 )
 
-class ZMQTest (BitcoinTestFramework):
+class Subscriber():
+    def __init__(self, socket, address):
+        self.address = address
+        self.sequence = 0
+        self.socket = socket
+
+class ZMQTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 2
+        self.subscribers = {}
+
+    def subscribe(self, type):
+        import zmq
```

Comment:
Why not move this logic into `ZMQSubscriber.__init__()`? Then you'd just need to pass in `self.zmqContext`. Port could be taken from a class variable to ensure uniqueness.

Reply:
`ZMQSubscriber` is like a C++ POD, did't want to have logic there. Is this a strong suggestion?

Reply:
It makes more sense to me to have `ZMQSubscriber` handle the initialization logic (and also preferably the receive logic). That makes it more portable if for some reason we want to use the interface in another test.

Conceptually it doesn't make sense to me to have a TestFramework object subscribing to a zmq interface. Owning ZMQ subscribers makes more sense to me.